presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -113,4 +122,31 @@ private static SymbolStatsEstimate addColumnStats(SymbolStatsEstimate leftStats,
                 .setNullsFraction((nullsCountLeft + nullsCountRight) / newRowCount)
                 .build();
     }
+
+    public static PlanNodeStatsEstimate intersect(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right)
+    {
+        checkArgument(new HashSet<>(left.getSymbolsWithKnownStatistics()).equals(new HashSet<>(right.getSymbolsWithKnownStatistics())));
```

Comment:
If I understand correctly this seems to be wrong. It is possible scenario that you intersect data from other connectors ones with stats and ones without. It can return UNKNOWN_STATS, but crashing seems like bad idea.